lib/Dialect/HW/ModuleImplementation.cpp

```diff
@@ -70,6 +70,11 @@ parseFunctionResultList(OpAsmParser &parser,
 
 /// Return the port name for the specified argument or result.
 static StringRef getModuleArgumentName(Operation *module, size_t argNo) {
+  if (auto mod = dyn_cast<HWModuleLike>(module)) {
```

Comment:
Okay, for HWModuleLike's use its helpers (similarly indicating out-of-range with empty name), otherwise look to the old attribute arrays.